### PR TITLE
Document metadata labels

### PR DESCRIPTION
**Pull request by KyleJu**
Add documentation for the `label` feature per https://github.com/web-platform-tests/wpt.fyi/issues/1491. See discussions [here](https://github.com/web-platform-tests/wpt.fyi/pull/2678#issuecomment-970365882)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +21,13 @@ Where
 - `product` is a browser name with an option of [product spec](https://github.com/web-platform-tests/wpt.fyi/blob/master/api/README.md)
   - `{browser-name}[-{browser-version}[-{os}[-{os-version}]]]`
   - e.g. `chrome`, `safari-12`, or `firefox-63.0-linux`
+  - when `product` is omitted, this `link` applies to tests across all browsers
+- `URL` is a bug URL; `URL` is non-optional unless a `label` is present.
```
Comment (foolip) [inline README.md:25]:
```suggestion
- `url` is a bug URL; `url` is non-optional unless a `label` is present.
```

File: README.md [hunk 2]
```diff
@@ -20,6 +21,13 @@ Where
...
+- `label` is a label to its tests; `label` is used at a test-level and do not
+  apply to subtests. When `label` is present in a link:
+  - `product`should be omitted
+  - `URL` is optional
```
Comment (foolip) [inline README.md:29]:
```suggestion
  - `url` is optional
```

File: README.md [hunk 3]
```diff
@@ -20,6 +21,13 @@ Where
...
+  - `Subtest name` is omitted
```
Comment (foolip) [inline README.md:30]:
```suggestion
  - `subtest` is omitted
```

File: README.md [hunk 4]
```diff
@@ -20,6 +21,13 @@ Where
...
 - `Test name` is a filename, which is relative to the current directory. If it
```
Comment (foolip) [inline README.md:31]: This should be `test` I think. (Existing issue.)

File: README.md [hunk 5]
```diff
@@ -20,6 +21,13 @@ Where
 - `product` is a browser name with an option of [product spec](https://github.com/web-platform-tests/wpt.fyi/blob/master/api/README.md)
   - `{browser-name}[-{browser-version}[-{os}[-{os-version}]]]`
   - e.g. `chrome`, `safari-12`, or `firefox-63.0-linux`
+  - when `product` is omitted, this `link` applies to tests across all browsers
+- `URL` is a bug URL; `URL` is non-optional unless a `label` is present.
+- `label` is a label to its tests; `label` is used at a test-level and do not
+  apply to subtests. When `label` is present in a link:
+  - `product`should be omitted
```
Comment (foolip) [inline README.md:28]:
```suggestion
  - `product` should be omitted
```

**Review by foolip — APPROVED** [review]
LGTM with typo fixes.

File: README.md (hunk 4 above)
Comment (KyleJu) [reply README.md:31]:
```suggestion
- `test` is a filename, which is relative to the current directory. If it
```

File: README.md [hunk 6]
```diff
@@ -20,7 +21,14 @@ Where
...
+- `url` is a bug URL; `url` is non-optional unless a `label` is present.
+- `label` is a label to its tests; `label` is used at a test-level and do not
+  apply to subtests. When `label` is present in a link:
+  - `product` should be omitted
+  - `url` is optional
+  - `subtest` is omitted
+- `test` is a filename, which is relative to the current directory. If it
   is `"*"` (note that asterisks must be quoted in YAML), the link will apply to
   all tests in the current directory and its subdirectories.
 - `Test result status` is an optional field that records the WPT test result, as
```
Comment (KyleJu) [inline README.md:34]:
```suggestion
- `status` is an optional field that records the WPT test result, as 
```

**Pull request merged**